lib/features/auth/presentation/widgets/security_settings.dart

```diff
@@ -10,201 +10,266 @@ import 'package:dairy_app/features/auth/presentation/bloc/user_config/user_confi
+                               .fingerPrintAthShouldBeEnabledInDeviceSettings,
+                           style: TextStyle(color: mainTextColor),
+                         ),
+                         value: isFingerPrintLoginEnabledValue ?? false,
+                         onChanged: (value) async {
+                           if (userId == null) {
+                             showToast(S.current.unexpectedErrorOccured);
+                             return;
+                           }
+
+
+                           if (userId == GuestUserDetails.guestUserId) {
+                             showToast(S.current
+                                 .pleaseSetupYourAccountToUseThisFeature);
+                             return;
+                           }
+
+
+                           try {
+                             await authenticationRepository
+                                 .isFingerprintAuthPossible();
+                             userConfigCubit.setUserConfig(
+                               UserConfigConstants.isFingerPrintLoginEnabled,
+                               value,
+                             );
+                           } on Exception catch (e) {
+                             showToast(
+                                 e.toString().replaceAll("Exception: ", ""));
+                           }
+                         },
+                       ),
+                       SwitchListTile(
+                         inactiveTrackColor: inactiveTrackColor,
+                         activeColor: activeColor,
+                         contentPadding: const EdgeInsets.all(0.0),
+                         title: Text(
+                           S.current.enablePINLogin,
+                           style: TextStyle(color: mainTextColor),
+                         ),
+                         subtitle: Text(
+                                    S.current
+                                        .pinLoginSetupInstructions,
+                                    style: TextStyle(color: mainTextColor),
+                         ),
+                        value: isPINLoginEnabledValue ?? false,
+                         onChanged: (value) async {
+                           if (userId == null) {
+                             showToast(S.current.unexpectedErrorOccured);
+                             return;
+                           }
+
+
+                           if (userId == GuestUserDetails.guestUserId) {
+                             showToast(S.current
+                                 .pleaseSetupYourAccountToUseThisFeature);
+                             return;
+                           }
+                           if (value) {
+                             // Call the PIN reset popup
+                             await pinResetPopup(
+                               context: context,
+                               userPinId: userId,
+                             );
+                           }
+
```

Comment:
This switch should be enabled only if PIN is successfully set, currently it gets enabled if we enter mismatched pins